sql/builtin_geo_functions.cc

```diff
@@ -0,0 +1,72 @@
+/** builtin_geo_functions.cc
+    Jeremy Barnes, 14 June 2015
+    This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
+
+    Builtin geometric functions for SQL.
+*/
+
+#include "mldb/sql/builtin_functions.h"
+#include "mldb/ext/s2/s2.h"
+#include "mldb/ext/s2/s2latlng.h"
+#include "mldb/types/basic_value_descriptions.h"
+
+
+using namespace std;
+
+
+namespace Datacratic {
+namespace MLDB {
+namespace Builtins {
+
+/*****************************************************************************/
+/* GEOGRAPHICAL FUNCTIONS                                                    */
+/*****************************************************************************/
+
+static constexpr double EARTH_EQUATORIAL_RADIUS_METERS = 6378137.0;
+static constexpr double EARTH_POLAR_RADIUS_METERS      = 6356752.3;
+
+// https://en.wikipedia.org/w/index.php?title=Earth_radius&action=edit&section=16
+static constexpr double EARTH_MEAN_RADIUS_METERS       = 6371008.8;
+
+BoundFunction geo_distance(const std::vector<BoundSqlExpression> & args)
+{
+    checkArgsSize(args.size(), 4);
+
+    auto outputInfo
+        = std::make_shared<Float64ValueInfo>();
+        
+    return {[=] (const std::vector<ExpressionValue> & args,
+                 const SqlRowScope & scope) -> ExpressionValue
+            {
+                ExcAssertEqual(args.size(), 4);
+
+                Date ts = calcTs(args[0], args[1], args[2], args[3]);
+                
+                if (args[0].empty() || args[1].empty()
+                    || args[2].empty() || args[3].empty())
+                    return ExpressionValue::null(ts);
+                
+                double lat1 = args[0].getAtom().toDouble();
```

Comment:
just because geodesy is a minefield, we should specify the datum that should be used here... Is there some underlying assumption that the input is in the WGS 84 coordinate system? does it work for any coordinate system?